swap/src/asb/config.rs

```diff
@@ -118,30 +177,11 @@ pub fn read_config(config_path: PathBuf) -> Result<Result<Config, ConfigNotIniti
     Ok(Ok(file))
 }
 
-/// Default location for storing the config file for the ASB
-// Takes the default system config-dir and adds a `/asb/config.toml`
-pub fn default_config_path() -> Result<PathBuf> {
-    system_config_dir()
-        .map(|dir| Path::join(&dir, "asb"))
-        .map(|dir| Path::join(&dir, "config.toml"))
-        .context("Could not generate default config file path")
-}
-
-/// Default location for storing data for the CLI
-// Takes the default system data-dir and adds a `/asb`
-fn default_data_dir() -> Result<PathBuf> {
-    system_data_dir()
-        .map(|proj_dir| Path::join(&proj_dir, "asb"))
-        .context("Could not generate default data dir")
-}
-
-pub fn initial_setup<F>(config_path: PathBuf, config_file: F) -> Result<()>
+pub fn initial_setup<F>(config_path: PathBuf, config_file: F, testnet: bool) -> Result<()>
 where
-    F: Fn() -> Result<Config>,
+    F: Fn(bool) -> Result<Config>,
 {
-    info!("Config file not found, running initial setup...");
-    let initial_config = config_file()?;
-
+    let initial_config = config_file(testnet)?;
```

Comment:
This seems overly complicated. You are passing two parameters in: `config_file` and `testnet` only to combine them as the very first thing to the `initial_config` variable. Why can't the caller just pass in `initial_config`?